[xpu][fix][Inductor] Catch Intel Triton compilation/runtime error as IntelGPUError.

etaf opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.14.0) (oldest at bottom):
* __->__ #169167

This PR updates the pinned Intel Triton commit (based on the 3.7.x release) to introduce IntelGPUError, which is used to catch compilation and runtime errors. This allows Inductor to safely discard Triton configurations that are not suitable for Intel GPUs.

Background:
The test case python test/inductor/test_native_matmul.py -k test_3mm_add
produces the following error on XPU:
```
  File "/workspacetriton/python/triton/compiler/compiler.py", line 489, in _init_handles
    self.module, self.function, self.n_regs, self.n_spills, self.n_max_threads = driver.active.utils.load_binary(
  File "/workspace/triton/python/triton/backends/intel/driver.py", line 217, in load_binary
    return self.shared_library.load_binary(args)
torch._inductor.exc.InductorError: RuntimeError: ZE_RESULT_ERROR_INVALID_KERNEL_NAME 
```
This occurs for one of the Triton kernel configs that hits a hardware limitation, causing Triton’s C++ code to raise PyExc_RuntimeError when compiling the SPIR-V kernel. Intel Triton has catched such Cpp error and raised to Inductor as `IntelGPUError` so that we can safely discard the failing config and continue other configs.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

etaf committed:
[xpu][fix][Inductor] Catch Triton Cpp raised PyExc_RuntimeError so that Inductor
can discard Triton configurations that can not be compiled or loaded
successfully.

[ghstack-poisoned]

EikanWang commented on torch/_inductor/runtime/triton_heuristics.py:
```diff
@@ -602,7 +602,12 @@ def _make_launchers(self):
+                    OutOfResources,
+                    PTXASError,
+                    torch.cuda.OutOfMemoryError,
+                    RuntimeError,  # raised in Triton C++ with PyExc_RuntimeError
```
@etaf , per my understanding, `RuntimeError` is too general to skip some exception that we want to raise. I'm considering asking the Triton team help to throw another execption dedicated to error.

etaf committed:
Update on "[WIP][xpu][fix][Inductor] Catch Triton Cpp raised PyExc_RuntimeError"


Catch Triton C++-raised `PyExc_RuntimeError` so that Inductor can discard Triton configurations that cannot be compiled or loaded successfully.

The test case python test/inductor/test_native_matmul.py -k test_3mm_add
produces the following error on XPU:
```
  File "/workspacetriton/python/triton/compiler/compiler.py", line 489, in _init_handles
    self.module, self.function, self.n_regs, self.n_spills, self.n_max_threads = driver.active.utils.load_binary(
  File "/workspace/triton/python/triton/backends/intel/driver.py", line 217, in load_binary
    return self.shared_library.load_binary(args)
torch._inductor.exc.InductorError: RuntimeError: ZE_RESULT_ERROR_INVALID_KERNEL_NAME 
```
This occurs for one of the Triton kernel configs that hits a hardware limitation, causing Triton’s C++ code to raise PyExc_RuntimeError when compiling the SPIR-V kernel. We need to catch this exception so Inductor can safely discard the failing config and continue other configs.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

etaf committed:
Update on "[WIP][xpu][fix][Inductor] Catch Triton Cpp raised PyExc_RuntimeError"


Catch Triton C++-raised `PyExc_RuntimeError` so that Inductor can discard Triton configurations that cannot be compiled or loaded successfully.

The test case python test/inductor/test_native_matmul.py -k test_3mm_add
produces the following error on XPU:
```
  File "/workspacetriton/python/triton/compiler/compiler.py", line 489, in _init_handles
    self.module, self.function, self.n_regs, self.n_spills, self.n_max_threads = driver.active.utils.load_binary(
  File "/workspace/triton/python/triton/backends/intel/driver.py", line 217, in load_binary
    return self.shared_library.load_binary(args)
torch._inductor.exc.InductorError: RuntimeError: ZE_RESULT_ERROR_INVALID_KERNEL_NAME 
```
This occurs for one of the Triton kernel configs that hits a hardware limitation, causing Triton’s C++ code to raise PyExc_RuntimeError when compiling the SPIR-V kernel. We need to catch this exception so Inductor can safely discard the failing config and continue other configs.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

jansel commented on .ci/docker/ci_commit_pins/triton-xpu.txt:
```diff
@@ -1 +1 @@
-64bb0de394fe1b9e163758380ce8eed25511bb5e
+307748db7742a0f8259a7ea0336909eb55d2051a
```
Is this update intentional?

etaf replied:
Hi @jansel, this is the latest commit of Initel Triton release 3.7.x branch.

etaf committed:
Update

[ghstack-poisoned]

etaf committed:
Update

[ghstack-poisoned]

etaf commented:
@pytorchbot merge
